x/stake/keeper.go

```diff
@@ -38,6 +39,18 @@ func (k Keeper) GetValidator(ctx sdk.Context, addr sdk.Address) (validator Valid
 	return k.getValidator(store, addr)
 }
 
+// get a single validator by pubkey
+func (k Keeper) GetValidatorByPubKey(ctx sdk.Context, pubkey crypto.PubKey) (validator Validator, found bool) {
+	store := ctx.KVStore(k.storeKey)
+	b := store.Get(GetValidatorByPubKeyKey(pubkey))
+	if b == nil {
+		return validator, false
+	}
+	var addr sdk.Address
+	k.cdc.MustUnmarshalBinary(b, &addr)
```

Comment:
address should not be marshalled - bytes from the store should already be the address

Reply:
Updated.